Forta-Bots-Challenge-3/src/utils.ts

```diff
@@ -0,0 +1,101 @@
+import { Finding } from "forta-agent";
+import { providers, Contract, BigNumber } from "ethers";
+import { createEscrowBalanceFinding, createInvariantViolationFinding } from "./finding";
+import { ERC20_ABI, BOT_ID, OPTIMISM_CHAIN_ID, ARBITRUM_CHAIN_ID, MAKER_ESCROW_ALERT_ID } from "./constants";
+
+export type escrowBalances = {
+  blockNumber: number;
+  optimismEscrowBalance: BigNumber | null;
+  arbitrumEscrowBalance: BigNumber | null;
+};
+
+export type contractAddresses = {
+  l1DaiAddress: string;
+  l2DaiAddress: string;
+  optimismEscrow: string;
+  arbitrumEscrow: string;
+};
+
+export const emitEscrowBalance = async (
+  provider: providers.Provider,
+  blockNumber: number,
+  l1DaiAddress: string,
+  optimismEscrow: string,
+  arbitrumEscrow: string,
+  prevEscrowBalance: escrowBalances,
+  findings: Finding[]
+) => {
+  const l1DaiContract = new Contract(l1DaiAddress, ERC20_ABI, provider);
+
+  // call L1 DAI contract to get balances of each L2's escrow contract
+  const optimismEscrowBalance = await l1DaiContract.balanceOf(optimismEscrow, { blockTag: blockNumber });
+  const arbitrumEscrowBalance = await l1DaiContract.balanceOf(arbitrumEscrow, { blockTag: blockNumber });
+
+  // create and send finding only if balances are not defined or have changed since last alert
+  if (
+    !prevEscrowBalance ||
+    prevEscrowBalance["optimismEscrowBalance"] !== optimismEscrowBalance ||
+    prevEscrowBalance["arbitrumEscrowBalance"] !== arbitrumEscrowBalance
```

Comment:
This doesn't work as expected with BigNumbers. You either need to convert them both to strings or use the BigNumber's `.eq()` operator.
```suggestion
    !prevEscrowBalance["optimismEscrowBalance"].eq(optimismEscrowBalance) ||
    !prevEscrowBalance["arbitrumEscrowBalance"] .eq(arbitrumEscrowBalance)
```